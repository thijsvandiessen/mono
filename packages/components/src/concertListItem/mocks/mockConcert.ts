import type { Concert } from '@mono/data'

export const mockConcert: Concert = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [
    {
      startTime: '2023-09-01T13:32:36+01:00',
      id: 'test id',
      title: 'test',
      address: 'test',
      ticketLink: 'https://example.com',
      lat: 0,
      lng: 0,
    },
  ],
  url: 'some-slug',
  content: [],
}
