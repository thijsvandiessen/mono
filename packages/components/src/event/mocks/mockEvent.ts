import type { Event } from '@mono/graphql'

export const mockEvent: Event = {
  id: 'id',
  title: 'name',
  image: {
    id: 'image-id',
    title: 'title',
    description: 'desc',
    url: 'https://creaorkest.nl/some-image.png',
  },
  locations: [{ startTime: '2023-09-01T13:32:36+01:00', id: 'test id' }],
  url: 'some-slug',
  content: [],
}
