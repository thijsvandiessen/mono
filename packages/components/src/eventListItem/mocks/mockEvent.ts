import type { Event } from '@mono/data'

export const mockEvent: Event = {
  id: 'id',
  title: 'name',
  image: undefined,
  locations: [{ startTime: '2023-09-01T13:32:36+01:00', id: 'test id' }],
  url: 'some-slug',
  content: [],
}
