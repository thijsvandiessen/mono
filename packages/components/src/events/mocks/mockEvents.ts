import type { Event } from '@mono/datocms'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  url: '/concerten/concert-1',
  image: undefined,
  locations: [],
}
