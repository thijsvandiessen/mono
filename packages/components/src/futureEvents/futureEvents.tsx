import React from 'react'
import Link from 'next/link'
import { EventListItem } from '../eventListItem'
import { getFutureEvents } from '@mono/data'
import type { Event } from '@mono/data'

export interface Props {
  skip: number
  first: number
}

export const FutureEvents = async ({ skip, first }: Props) => {
  const { data } = await getFutureEvents({ skip, first })

  const events = data.reduce((previousValues: Event[], currentvalue) => {
    const future = currentvalue?.locations.find((location) => {
      if (!location?.startTime) {
        return false
      }

      try {
        return new Date(location?.startTime ?? '') > new Date()
      } catch {
        return false
      }
    })

    if (future?.id) {
      return [...previousValues, currentvalue]
    }
    return previousValues
  }, [])

  if (!data || events?.length === 0)
    return (
      <>
        <h2>Komende concerten</h2>
        <p>
          Houd de website in de gaten om up to date te blijven over volgende
          concerten.
        </p>
        <Link href="/concerten">Bekijk al onze vorige concerten</Link>
      </>
    )

  return (
    <div>
      {data.map((event, index) => {
        if (!event?.id) return null
        return (
          <EventListItem
            className="content-layout--small"
            key={event.id}
            data={event}
            size="large"
            isLast={data.length - 1 === index}
          />
        )
      })}
    </div>
  )
}
