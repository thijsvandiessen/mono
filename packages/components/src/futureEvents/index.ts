export { FutureEvents } from './futureEvents.js'
