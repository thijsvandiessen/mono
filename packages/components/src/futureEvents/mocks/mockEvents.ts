import { type Event } from '@crea/graphql/src/types/event'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  image: undefined,
  locations: [],
  persons: [],
  program: [],
  url: '/some-slug',
  content: [],
}
