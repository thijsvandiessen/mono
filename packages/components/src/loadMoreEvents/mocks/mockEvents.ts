import { type Event } from '@mono/datocms'

export const mockEvent: Event = {
  id: 'mock-id',
  title: 'title',
  url: '/url',
  image: undefined,
  locations: [],
}
