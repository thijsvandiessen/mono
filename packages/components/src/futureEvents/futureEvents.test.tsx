import type { Props as EventListItemProps } from '../eventListItem'
import { FutureEvents } from './futureEvents'
import type { Props } from './futureEvents'
import React from 'react'
import { getFutureEvents } from '@crea/graphql/src/getters/getFutureEvents'
import { mockEvent } from './mocks/mockEvents'
import { resolvedComponent } from '@crea/utils/src/testHelpers/resolvedComponent'
import { render, screen } from '@testing-library/react'

jest.mock('@crea/graphql/src/getters/getFutureEvents', () => {
  const originalModule = jest.requireActual(
    '@crea/graphql/src/getters/getFutureEvents'
  )
  return {
    __esModule: true,
    ...originalModule,
    getFutureEvents: jest.fn(),
  }
})

jest.mock('../eventListItem/eventListItem', () => {
  const originalModule = jest.requireActual('../eventListItem/eventListItem')
  return {
    __esModule: true,
    ...originalModule,
    EventListItem: ({ data }: EventListItemProps) => (
      <div>
        [Event]: <span>{data.id}</span>
      </div>
    ),
  }
})

const getFutureEventsMock = jest.mocked(getFutureEvents)

describe('FutureEvents component', () => {
  it('shows all the data', async () => {
    getFutureEventsMock.mockResolvedValue({
      data: [mockEvent],
      error: undefined,
    })

    const ResolvedFutureEvents = await resolvedComponent<Props>(FutureEvents, {
      skip: 0,
      first: 3,
    })

    const { container } = render(<ResolvedFutureEvents />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('mock-id')).toBeInTheDocument()
  })
})
