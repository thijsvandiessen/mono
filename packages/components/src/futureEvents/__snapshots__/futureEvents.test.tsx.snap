// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FutureEvents component > shows all the data 1`] = `
<div>
  <div>
    <div>
      <span>
        mock-id
      </span>
    </div>
  </div>
</div>
`;

exports[`FutureEvents component > shows no data 1`] = `
<div>
  <h2>
    Komende concerten
  </h2>
  <p>
    Houd de website in de gaten om up to date te blijven over volgende concerten.
  </p>
  <a
    href="/concerten"
  >
    Bekijk al onze vorige concerten
  </a>
</div>
`;
